### PR TITLE
Use jll package

### LINKED ISSUE #7: Add LAPACK binary dependency

### LINKED ISSUE #14: Cannot create circles with Gmsh binary bundled with package
Trying to use 

```
gmsh.model.occ.addCircle(0.05, 0.15, 0, 0.01, 2)
```

gives

```
Error   : Gmsh requires OpenCASCADE to add circle
ERROR: LoadError: gmshModelOccAddCircle returned non-zero error code: 1
```

when using `Gmsh`. If I download the official Gmsh and use the `gmsh.jl` in the `lib` folder, everything works fine.

### LINKED ISSUE #20: troubles in running occ functions
Hi,

I am having troubles using occ functions, for example I was trying to run the example t16 from here [t16.jl](https://gitlab.onelab.info/gmsh/gmsh/-/blob/master/tutorial/julia/t16.jl) but I got this error

`import Gmsh: gmsh`
`gmsh.initialize()`
`gmsh.model.add("t16")`
`gmsh.model.occ.addBox(0,0,0, 1,1,1, 1)`

`gmshModelOccAddBox returned non-zero error code: 1`

could you please help me finding out what am I doing wrong here?

many thanks,
Andrea

### PR DESCRIPTION
**Pull request by JonasIsensee**
Hi,

I'm not sure if there's a consensus for the future of this package.
However, since I wanted to use the `Gmsh` API, I changed things to work the same as in `GridapGmsh.jl` which uses the JLL package.

With these changes, it works for me locally on ubuntu.

closes #14 
closes #7
probably fixes #20 (at least for me locally)

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -1,37 +1,47 @@
...
+  else
+    @warn """
+    gmsh Julia API not found in $gmsh_jl
+    Make sure that you have installed the full gmsh SDK"""
+  end
 end
 
-# Write out a deps.jl file that will contain mappings for our products
-write_deps_file(joinpath(@__DIR__, "deps.jl"), products, verbose=verbose)
+open("deps.jl","w") do f
```
Comment (koehlerson) [inline deps/build.jl:41]: what is the purpose of this? Is this just for debugging stuff?

Comment (koehlerson) [reply deps/build.jl:41]: ok nvm just found it, is there are a reason why this is still written to a file? I actually would prefer to just have the variables `GMSH_FOUND` and `gmsh_jl` inside `src/Gmsh.jl`

File: src/Gmsh.jl [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
-global const api = gmsh
+using Libdl
 
-function __init__()
-    # Always check your dependencies that live in `deps.jl`
-    check_deps()
+deps_jl = joinpath(@__DIR__, "..", "deps", "deps.jl")
+if !isfile(deps_jl)
+  s = """
+  Package GridapGmsh not installed properly.
```
Comment (koehlerson) [inline src/Gmsh.jl]:
```suggestion
  Package Gmsh.jl not installed properly.
```

File: src/Gmsh.jl [hunk 3]
```diff
@@ -1,18 +1,44 @@
...
+  Run Pkg.build(\"GridapGmsh\"), restart Julia and try again.
```
Comment (koehlerson) [inline src/Gmsh.jl]:
```suggestion
  Run Pkg.build(\"Gmsh\"), restart Julia and try again.
```

File: src/Gmsh.jl [hunk 4]
```diff
@@ -1,18 +1,44 @@
...
 end
 
+include(deps_jl)
 
+if GMSH_FOUND
+    include(gmsh_jl)
+    # Hack taken from MPI.jl
+    function __init__()
+      @static if Sys.isunix()
+        Libdl.dlopen(gmsh.lib, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
```
Comment (koehlerson) [inline src/Gmsh.jl:24]: Can you explain what's happening here? I have the feeling things could be simplified a little: https://github.com/Ferrite-FEM/FerriteGmsh.jl/blob/master/src/FerriteGmsh.jl#L5-L7

File: src/Gmsh.jl [hunk 5]
```diff
@@ -1,18 +1,44 @@
...
+      end
+    end
+  else
+      s = """
+      Gmsh not found in system paths.
+      Install Gmsh or export path to Gmsh and rebuild the project.
```
Comment (koehlerson) [inline src/Gmsh.jl]:
```suggestion
      Renstall Gmsh.jl or export `GMSHROOT` that points to a Gmsh installation and rebuild the project.
```

File: src/Gmsh.jl [hunk 6]
```diff
@@ -1,18 +1,44 @@
...
+      Run Pkg.build(\"GridapGmsh\"), restart Julia and try again.
```
Comment (koehlerson) [inline src/Gmsh.jl]:
```suggestion
      Run Pkg.build(\"Gmsh\"), restart Julia and try again.
```

File: src/Gmsh.jl [hunk 7]
```diff
@@ -1,18 +1,44 @@
...
+      """
+      @warn s
+  end
+  
+  macro check_if_loaded()
+    quote
+      if ! GMSH_FOUND
+        error("GridapGmsh is not loaded or installed properly.")
```
Comment (koehlerson) [inline src/Gmsh.jl]:
```suggestion
        error("Gmsh is not loaded or installed properly.")
```

**Comment by koehlerson** [comment]
Sorry for the delayed reply, kinda forgot about this repo

**Comment by fredrikekre** [comment]
There are now new and shiny ways to override things from the JLL package, maybe that can be used instead? See https://docs.binarybuilder.org/stable/jll/#Overriding-the-artifacts-in-JLL-packages

File: src/Gmsh.jl (hunk 4 above)
Comment (JonasIsensee) [reply src/Gmsh.jl:24]: Interesting. No, I can't explain.
But if your much shorter code works that's surely preferable.

**Comment by JonasIsensee** [comment]
Hi @koehlerson,
no worries.
About your questions: tbh I had no idea what I was doing and as you will have noticed, I essentially copied the code from 
https://github.com/gridap/GridapGmsh.jl/blob/master/deps/build.jl

I'm sure there's an easier way.

**Comment by koehlerson** [comment]
closed in favor for https://github.com/JuliaFEM/Gmsh.jl/pull/22

**Pull request closed without merging**